plugins/doodles/src/App.tsx

```diff
@@ -0,0 +1,345 @@
+import { useState, useRef } from "react";
+import { framer } from "framer-plugin";
+import { ReactSketchCanvas, ReactSketchCanvasRef } from "react-sketch-canvas";
+import * as Slider from "@radix-ui/react-slider";
+import "./App.css";
+
+function handleFocus(event: React.FocusEvent<HTMLInputElement>) {
+  event.target.select();
+}
+
+const drawStyles = {
+  border: "none",
+  backgroundColor: "transparent",
+};
+
+async function svgToBytes(svgText: string) {
+  const blob = new Blob([svgText], { type: "image/svg+xml" });
+  const arrayBuffer = await new Promise<ArrayBuffer>((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = () => resolve(reader.result as ArrayBuffer);
+    reader.onerror = reject;
+    reader.readAsArrayBuffer(blob);
+  });
+
+  const uint8Array = new Uint8Array(arrayBuffer);
+
+  return uint8Array;
+}
+
+export async function getAssetDataFromSVG(svgString: string): Promise<string> {
+  const encodedSvg = encodeURIComponent(svgString);
+  const imageUrl = `data:image/svg+xml;charset=utf-8,${encodedSvg}`;
+  return imageUrl;
+}
+export function App() {
+  const handleAddSvg = async (drawing: string) => {
+    await framer.addImage({
+      image: {
+        type: "bytes",
+        bytes: await svgToBytes(drawing),
+        mimeType: "image/svg+xml",
+      },
+      name: "Doodle",
+    });
+  };
+
+  const canvasRef = useRef<ReactSketchCanvasRef>(null);
+  const strokeInputRef = useRef<HTMLInputElement>(null);
+  const strokeHueInputRef = useRef<HTMLInputElement>(null);
+
+  const [strokeValue, setStrokeValue] = useState(5);
+  const [strokeInputValue, setStrokeInputValue] = useState("5");
+
+  const [strokeHueInputValue, setStrokeHueInputValue] = useState("0");
+  const [strokeSaturateInputValue, setStrokeSaturateInputValue] = useState("0");
+  const [strokeLightInputValue, setStrokeLightInputValue] = useState("50");
```

Comment:
Is there a reason to have separate states for the input and rendered values? They seem to always be in sync

Reply:
I think so, because these inputs mimic the framer ux of allowing you to type in the input and it not insta setting the value until you hit enter. Otherwise values would update as you type?

Reply:
ahh I see, that makes sense